lib/active_triples/persistence_strategies/parent_strategy.rb

```diff
@@ -87,6 +85,27 @@ def reload
       true
     end
 
+    private
+
+    def enum_ancestors
+      raise NilParentError if parent.nil?
+
+      Enumerator.new do |yielder|
```

Comment:
Why not define a ruby class for enumerating over ancestors and use that, rather than an in-line Enumerator?